homeassistant/components/emoncms/sensor.py

```diff
@@ -115,47 +122,52 @@ async def async_setup_platform(
         sensors.append(
             EmonCmsSensor(
                 hass,
-                emoncms_client,
+                coordinator,
                 name,
                 value_template,
                 unit_of_measurement,
                 str(sensorid),
-                elem,
+                idx,
             )
         )
     async_add_entities(sensors)
 
 
-class EmonCmsSensor(SensorEntity):
+class EmonCmsSensor(
+    CoordinatorEntity[DataUpdateCoordinator[list[dict[str, Any]] | None]], SensorEntity
+):
     """Implementation of an Emoncms sensor."""
 
     def __init__(
         self,
         hass: HomeAssistant,
```

Comment:
For a follow up, you don't need to set `self._hass` after the entity got added to home assistant, you can access the object in `self.hass`

Reply:
Thanks, so i remove as self._hass is not used actually in EmonCmsSensor...was there before, I forgot to remove